### DISCUSSION TITLE
Prevent self-hosted runners accepting new jobs

### DISCUSSION THREAD

**Discussion by sjackson-ccdc** [community/community#123082] [Actions]
### Select Topic Area

Product Feedback

### Body

We heavily use self-hosted runners in our organisation, but sometimes we need to run system maintenance or upgrades that require us shutting down various runner groups. It would be great if in the GitHub UI there was a feature that would prevent runners from accepting new jobs but allowing them to finish the current job. This would allow us to plan maintenance without disturbing ongoing workflows.

**Comment by crohr** [comment]
Is there a reason for not using ephemeral self-hosted runners like with [ARC](https://github.com/actions/actions-runner-controller) or [RunsOn](https://runs-on.com)?

Or another idea: make use of the [pre/post scripts](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/running-scripts-before-or-after-a-job) to make the runner shut down or perform maintenance immediately after finishing a specific workflow job? (you have access to the workflow name etc. in those scripts, so you could have a `maintenance` workflow that triggers that script when it is run)
